vllm/v1/core/sched/scheduler.py

```diff
@@ -434,6 +456,13 @@ def schedule(self) -> SchedulerOutput:
             grammar_bitmask=grammar_bitmask,
         )
 
+        # NOTE(Kuntai): this function is designed for multiple purposes:
+        # 1. Plan the KV cache store
+        # 2. Wrap up all the KV cache load / save ops into an opaque object
+        # 3. Clear the internal states of the connector
+        if self.connector is not None:
+            self.connector.attach_connector_meta(scheduler_output)
```

Comment:
Its confusing that `attach_connector_meta` mutates `scheduler_output`

I think we should update this to be:
- `scheduler_output = self.connector.attach_connector_meta(scheduler_output)` 

This will make it clear we are mutating scheduler output.